### DISCUSSION TITLE
VB.Net application won't close after creating a scheduler job

### DISCUSSION THREAD

**Discussion by schmeed** [quartznet/quartznet#454] [General]
First, I had to DIG HARD to find VB.NET examples for Quartz.Net.  I know it's not the preferred language but it would be nice if you would document other languages.

To the root of my issue:
Whenever I add a scheduled job (occur at a future date/time), it works fine.  However, when I try to close the application (i.e. file menu, close button, etc.) it won't close.

In Visual Studio, the app stays in run mode.  Outside of VS, the process hangs around after closing the main form.

Do I need to destroy all the objects?  Or is something else wrong?

Thanks!

**Comment by schmeed** [comment]
I added this to my close event:
sched.Shutdown() and now the application closes properly.

I believe this is what I need to do, but would like confirmation.

**Comment by lahma** [comment]
Yes, you probably had threads running that didn't exit until ready, shutting down scheduler cleans up all background threads that would prohibit closing the software.
